src/main/java/org/assertj/core/internal/Iterables.java

```diff
@@ -1086,6 +1087,21 @@ public void assertContainsExactly(AssertionInfo info, Iterable<?> actual, Object
     }
   }
 
+  public <E> void assertNoneSatisfy(AssertionInfo info, Iterable<? extends E> actual, Consumer<? super E> requirements) {
+    assertNotNull(info, actual);
+    requireNonNull(requirements, "The Consumer<T> expressing the assertions requirements must not be null");
+
+    for(E element : actual) {
+      try {
+        requirements.accept(element);
```

Comment:
if we pass this line, it means `element` is faulty, let's store it in a local variable to put it in the error message.